build.sbt

```diff
@@ -0,0 +1,113 @@
+import com.typesafe.sbt.packager.Keys.{
+  daemonUser,
+  defaultLinuxInstallLocation,
+  dockerAlias,
+  dockerBaseImage,
+  dockerCmd,
+  dockerEntrypoint,
+  dockerUpdateLatest,
+  maintainer,
+  packageName
+}
+import com.typesafe.sbt.packager.docker.{Cmd, DockerAlias, ExecCmd}
+
+name := "codacy-stylelint"
+
+version := "1.0.0-SNAPSHOT"
+
+val languageVersion = "2.12.4"
+
+scalaVersion := languageVersion
+
+resolvers ++= Seq(
+  "Sonatype OSS Snapshots" at "https://oss.sonatype.org/content/repositories/releases",
+  "Typesafe Repo" at "http://repo.typesafe.com/typesafe/releases/")
+
+libraryDependencies ++= Seq(
+  "com.codacy" %% "codacy-engine-scala-seed" % "3.0.183",
+  "com.vladsch.flexmark" % "flexmark-all" % "0.34.8",
+  "org.specs2" %% "specs2-core" % "4.2.0" % Test)
+
+scalacOptions ++= Common.compilerFlags
+scalacOptions.in(Compile, console) --= Seq("-Ywarn-unused", "-Ywarn-unused:imports", "-Xfatal-warnings")
+scalacOptions in Test ++= Seq("-Yrangepos")
+enablePlugins(JavaAppPackaging)
+
+enablePlugins(DockerPlugin)
+
+version in Docker := "1.0.0-SNAPSHOT"
+
+organization := "com.codacy"
+
+lazy val toolVersion = taskKey[String]("Retrieve the version of the underlying tool from patterns.json")
+toolVersion := {
+  import better.files.File
+  import play.api.libs.json.{JsString, JsValue, Json}
+
+  val jsonFile = resourceDirectory.in(Compile).value / "docs" / "patterns.json"
+  val patternsJsonValues = Json.parse(File(jsonFile.toPath).contentAsString).as[Map[String, JsValue]]
+
+  patternsJsonValues.collectFirst {
+    case ("version", JsString(ver)) => ver
+  }.getOrElse(throw new Exception("Failed to retrieve version from docs/patterns.json"))
+}
+
+mappings.in(Universal) ++= resourceDirectory
+  .in(Compile)
+  .map { resourceDir: File =>
+    val src = resourceDir / "docs"
+    val dest = "/docs"
+
+    (for {
+      path <- better.files.File(src.toPath).listRecursively()
+      if !path.isDirectory
+    } yield path.toJava -> path.toString.replaceFirst(src.toString, dest)).toSeq
+  }
+  .value
+
+def installAll(toolVersion: String) =
+  s"""apk update &&
+     |apk add bash curl nodejs-npm &&
+     |npm install -g npm@5 &&
```

Comment:
This needs to also be installed in CircleCI.
Move this to a install.sh file or use a package.json and do npm install.
This should be the same in the docker and on CircleCI.